Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1458,16 +1458,25 @@ private extension ChatChannelController {
     }
 
     private func getFirstUnreadMessageId() -> MessageId? {
-        guard let lastReadMessageId = channel?.reads.first(where: { $0.user.id == client.currentUserId })?.lastReadMessageId else {
-            // We default to the oldest message in the history
-            return messages.last?.id
```

Comment:
Why do we return this here? @polqf 

I don't get the logic behind these two tests:
- test_firstUnreadMessageId_whenReadsDoesNotContainCurrentUserId()
- test_firstUnreadMessageId_whenReadsContainsCurrentUserId_whenLastReadMessageIdIsNil()

For me, it looks like it should return `nil`. If we can't get the `lastReadMessageId()` it means there are no unreads, right?

Reply:
> If we can't get the lastReadMessageId it means there are no unreads

It is the other way around. If there are no "reads", it means the first unread message is the first one in the history

Reply:
How so? We now filter the lastReadMessageId based on the unreadCount. If there are no unread counts and no lastReadMessageId, it means there are no unreads. So how do we default to the first message in history?